modules/databend/databend.go

```diff
@@ -0,0 +1,139 @@
+package databend
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	databendUser        = "databend"
+	defaultUser         = "databend"
+	defaultPassword     = "databend"
+	defaultDatabaseName = "default"
+)
+
+// DatabendContainer represents the Databend container type used in the module
+type DatabendContainer struct {
+	testcontainers.Container
+	username string
+	password string
+	database string
+}
+
+func WithDefaultCredentials() testcontainers.CustomizeRequestOption {
+	return func(req *testcontainers.GenericContainerRequest) error {
+		username := req.Env["QUERY_DEFAULT_USER"]
+		password := req.Env["QUERY_DEFAULT_PASSWORD"]
+		if len(password) == 0 {
+			req.Env["QUERY_DEFAULT_PASSWORD"] = defaultPassword
+		}
+		if len(username) == 0 {
+			req.Env["QUERY_DEFAULT_USER"] = defaultUser
+		}
+		return nil
+	}
+}
+
+// Run creates an instance of the Databend container type
+func Run(ctx context.Context, img string, opts ...testcontainers.ContainerCustomizer) (*DatabendContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        img,
+		ExposedPorts: []string{"8000/tcp"},
+		Env: map[string]string{
+			"QUERY_DEFAULT_USER":     defaultUser,
+			"QUERY_DEFAULT_PASSWORD": defaultPassword,
+		},
+		WaitingFor: wait.ForLog("port: 8000  Databend Query Server"),
+	}
+
+	genericContainerReq := testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	}
+
+	opts = append(opts, WithDefaultCredentials())
+
+	for _, opt := range opts {
+		if err := opt.Customize(&genericContainerReq); err != nil {
+			return nil, err
+		}
+	}
+
+	username, ok := req.Env["QUERY_DEFAULT_USER"]
+	if !ok {
+		username = defaultUser
+	}
+	password := req.Env["QUERY_DEFAULT_PASSWORD"]
+
+	if len(password) == 0 && password == "" && !strings.EqualFold(defaultUser, username) {
+		return nil, fmt.Errorf("empty password and user")
+	}
+
+	container, err := testcontainers.GenericContainer(ctx, genericContainerReq)
+	var c *DatabendContainer
+	if container != nil {
+		c = &DatabendContainer{
+			Container: container,
+			password:  password,
+			username:  username,
+			database:  defaultDatabaseName,
```

Comment:
I add a `WithDatabase` method using defined a type to satisfy the testcontainers.ContainerCustomizer interface.

Reply:
You probably need a way to transfer the default options (db name) to the container request (as an env var??) so that the databend container uses that db name.

Reply:
That' why I define a custom type, there has no DB env in databend container. 😢

Reply:
Ok, understood. What I do not see is how the `WithDatabase` option is passing its value to the `container.database` field. In general, we usually do like this: https://github.com/testcontainers/testcontainers-go/blob/main/modules/redpanda/redpanda.go#L100-L109, defining a settings struct that will receive the configuration by the functional option, and then passing its values to the container.

Reply:
I'm unresolving this comment because otherwise the database setting won't be applied to the container. @hantmac I can assist you and add it myself if needed 😊

Reply:
The `WithDatabase` can't pass its value to the `container.database` directly, or though we pass it, it won't take effect until we execute `use newDatabase` or create another new connection dsn string to make a new connect.